File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
sys.setrecursionlimit(10**6)

result = 0

def dfs(graph, start, visited):
    visited[start] = True
    global result
    result += 1

    for g in graph[start]:
        if not visited[g]:
            dfs(graph, g, visited)

n = int(sys.stdin.readline().strip())
pairs = int(sys.stdin.readline().strip())
graph = [[] for _ in range(n+1)]
visited = [False for _ in range(n+1)]

for _ in range(pairs):
    a, b = map(int, sys.stdin.readline().strip().split())
    graph[b].append(a)
    graph[a].append(b)

dfs(graph, 1, visited)
print(result-1)

File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

void dfs(vector<vector<int>>& graph, vector<bool>& visited, int start) {
    visited[start] = true;

    for (int g: graph[start]) {
        if (!visited[g]) {
            dfs(graph, visited, g);
        }
    }
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int n, m;
    cin >> n >> m;

    vector<vector<int>> graph(n + 1);
    vector<bool> visited(n + 1, false);

    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        graph[u].emplace_back(v);
        graph[v].emplace_back(u);
    }

    dfs(graph, visited, 1);
    cout << count(visited.begin(), visited.end(), true) - 1 << "\n";

    return 0;
}
